ksql-engine/src/test/java/io/confluent/ksql/structured/SchemaKStreamTest.java

```diff
@@ -228,61 +229,115 @@ public void testSelectSchemaKStream() {
+
+    // When:
+    final SchemaKStream result = initialSchemaKStream
+        .select( selectExpressions, childContextStacker, processingLogContext);
+
+    assertThat(result.getKeyField(),
+        is(KeyField.of("NEWKEY", new Field("NEWKEY", 0, Schema.OPTIONAL_INT64_SCHEMA))));
+  }
 
+  @Test
+  public void shouldUpdateKeyIfRenamedViaFullyQualifiedName() {
+    // Given:
+    final PlanNode logicalPlan = givenInitialKStreamOf(
+        "SELECT test1.col0 as NEWKEY, col2, col3 FROM test1;");
+    final ProjectNode projectNode = (ProjectNode) logicalPlan.getSources().get(0);
     final List<SelectExpression> selectExpressions = projectNode.getProjectSelectExpressions();
-    final SchemaKStream projectedSchemaKStream = initialSchemaKStream.select(
-        selectExpressions,
-        childContextStacker,
-        processingLogContext);
-    assertThat(
-        projectedSchemaKStream.getKeyField(),
-        equalTo(Optional.of(new Field("NEWKEY", 0, Schema.OPTIONAL_INT64_SCHEMA))));
+
+    // When:
+    final SchemaKStream result = initialSchemaKStream
+        .select(selectExpressions, childContextStacker, processingLogContext);
+
+    // Then:
+    assertThat(result.getKeyField(),
+        is(KeyField.of("NEWKEY", new Field("NEWKEY", 0, Schema.OPTIONAL_INT64_SCHEMA))));
+  }
+
+  @Test
+  public void shouldUpdateKeyIfRenamedAndSourceIsAliased() {
+    // Given:
+    final PlanNode logicalPlan = givenInitialKStreamOf(
+        "SELECT t.col0 as NEWKEY, col2, col3 FROM test1 t;");
+    final ProjectNode projectNode = (ProjectNode) logicalPlan.getSources().get(0);
+    final List<SelectExpression> selectExpressions = projectNode.getProjectSelectExpressions();
+
+    // When:
+    final SchemaKStream result = initialSchemaKStream
+        .select(selectExpressions, childContextStacker, processingLogContext);
+
+    // Then:
+    assertThat(result.getKeyField(),
+        is(KeyField.of("NEWKEY", new Field("NEWKEY", 0, Schema.OPTIONAL_INT64_SCHEMA))));
   }
 
   @Test
   public void shouldPreserveKeyOnSelectStar() {
+    // Given:
     final PlanNode logicalPlan = givenInitialKStreamOf("SELECT * FROM test1;");
-
     final ProjectNode projectNode = (ProjectNode) logicalPlan.getSources().get(0);
     final List<SelectExpression> selectExpressions = projectNode.getProjectSelectExpressions();
-    final SchemaKStream projectedSchemaKStream = initialSchemaKStream.select(
-        selectExpressions,
-        childContextStacker,
-        processingLogContext);
-    assertThat(
-        projectedSchemaKStream.getKeyField(),
-        equalTo(initialSchemaKStream.getKeyField()));
+
+    // When:
+    final SchemaKStream result = initialSchemaKStream
+        .select(selectExpressions, childContextStacker, processingLogContext);
+
+    // Then:
+    assertThat(result.getKeyField(),
+        is(KeyField.of(Optional.of("COL0"), initialSchemaKStream.keyField.legacy())));
```

Comment:
why not `is(initialSchemaKStream.keyField)`?

Reply:
Because they're not currently the same. (It's on my todo list!)